### ISSUE TITLE
Add an option to 'remove torrent when finished' 

### ISSUE THREAD

**Issue by calexil** [transmission/transmission#217]
it's kinda silly this hasn't been implemented yet.

once a torrent has reached its desired share ratio there should be a checkbox to remove it.

This helps unclutter the downloads page for people who frequently cycle through seasons/etc

**Comment by Stibbons** [comment]
+1

**Comment by calexil** [comment]
was this ever fixed?

**Comment by buhtz** [comment]
I would say this is against "torrent netiquete".
This is a feature that shouldn't be implemented. But this is just my opinion.

Can you please describe more detailed a use-case where this feature would be helpfull?

**Comment by Stibbons** [comment]
When a torrent is "finished", it is no more seeded. So it can be removed. Of course this after the Seed ratio has been reached.

**Comment by calexil** [comment]
@buhtz when you have a couple dozen torrents seeding, and they reach the desired ratio. 

there is no need to keep them in torrent list cluttering up the interface. 

It is also an issue for people whose isp is shitty about seeding(even legally)

**Comment by h0ek** [comment]
+1

**Comment by Stibbons** [comment]
I now use a torrent-done.sh function executed at the end of the download. But this should a simple option in transmission

**Comment by NohaTech** [comment]
You can use flexget with the clean_transmission script for this.

**Comment by h0ek** [comment]
Wow, looks good. Thanks!

**Comment by calexil** [comment]
poke.

**Comment by calexil** [comment]
poke

**Comment by rodibin-at-placemaildotonline** [comment]
> I now use a torrent-done.sh function executed at the end of the download. But this should a simple option in transmission

Mind sharing your script here?

**Comment by h0ek** [comment]
@xelra
This could be connected with seed time. Something like: autoremove = after download complete, false, with data, after 1h of seed (or 1 day of seed etc. In my opinion seedtime and autoremove should be related.

**Comment by xelra** [comment]
@h0ek Yes, you're right. If the seed time feature were implemented, you could simply set `seedtime=0` and then torrent completion would invoke immediately.
In `settings.json` you would put
```
...
"script-torrent-done-enabled": true,
"script-torrent-done-filename": "/path/to/script.sh",
...
```
and the sript could look something like
```
#!/bin/bash

SERVER="localhost:9091 --auth user:pass"

transmission-remote $SERVER --torrent $TR_TORRENT_ID --remove-and-delete
```
@rodibin-at-placemaildotonline If you use this script right now, it's still gonna work, but torrent completion will only invoke either after the targeted ratio or seed-idle-time has been hit. I'm not sure whether it would work, if you set `seedratio=0`.

**Comment by afteroot** [comment]
Any updates for this ?

**Comment by Anan5a** [comment]
Any intention on implementing this feature?

**Comment by fssbchico** [comment]
Seconded ^

**Comment by Jesus-M** [comment]
+1

**Comment by cocoonkid** [comment]
this is really still missing?

**Comment by xeviff** [comment]
I'm also interested in it. But with deletion after X time

**Comment by xeviff** [comment]
> 

share it, please :)

**Comment by xeviff** [comment]
@gjalsem can you share your "torrent-done.sh"? :)

**Comment by killemov** [comment]
@xeviff On linux you could use sleep or, even better, cron to schedule some actions.

**Comment by xeviff** [comment]
oh, thanks. But at the end I migrated to qBittorrent which has a [tool](https://github.com/StuffAnThings/qbit_manage) for this kind of purpose. Not working fine for all my expectations, but it has many automations that solves most typical use cases.
